### PR TITLE
Rename (and change behaviour) of rule-no-shorthand-property-overrides

### PR DESCRIPTION
**Pull request by jeddy3**
Ref: https://github.com/stylelint/stylelint/issues/664

I [change the behaviour](https://github.com/stylelint/stylelint/pull/820/files#diff-d060e376b9d95e6ab025402d11db96beR21) of this rule to align it with `declaration-block-no-duplicate-properties` i.e. it now also checks `@rules` (for when they are nested).

I added a couple of tests to cover this.

### CODE REVIEW

File: src/rules/rule-no-shorthand-property-overrides/README.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # rule-no-shorthand-property-overrides
 
+**Deprecated: use the [`declaration-block-no-shorthand-property-overrides`](../declaration-block-no-shorthand-property-overrides/README.md) rule instead.**
+
```
Comment (davidtheclark) [inline src/rules/rule-no-shorthand-property-overrides/README.md]: Just double-checking ... did you remember to put this note at the top of the other readmes?

**Comment by davidtheclark** [comment]
Nice blitz, @jeddy3!

File: src/rules/rule-no-shorthand-property-overrides/README.md (hunk 1 above)
Comment (jeddy3) [inline src/rules/rule-no-shorthand-property-overrides/README.md]: I did (kind of). I forgot `rule-trailing-semicolon` first time around, and so I patched it in master.

**Comment by jeddy3** [comment]
> Nice blitz, @jeddy3!

No worries :) Shall I marked this as reviewed?

**Pull request merged**